pkg/apiserver/registry/core/shoot/strategy.go

```diff
@@ -239,6 +242,9 @@ func (shootStatusStrategy) PrepareForUpdate(_ context.Context, obj, old runtime.
 		gardencorehelper.ShouldETCDEncryptionKeyRotationBeAutoCompleteAfterPrepared(newShoot.Status.Credentials) {
 		newShoot.Generation = oldShoot.Generation + 1
 	}
+
+	// Migrate encryptedResources status from `.status.encryptedResources` to `status.credentials.encryptionAtRest.resources`.
```

Comment:
// Ensure that encrypted resources are synced from `.status.encryptedResources` to `status.credentials.encryptionAtRest.resources`.